fix tab accessibility issues on query building

fzhao99 opened the pull request:
# PULL REQUEST

## Summary

What changes are being proposed? For PRs that include UX/UI changes please make sure to describe their scope and highlight any specific areas that design reviews should focus.

## Related Issue

Fixes #

## Additional Information

Anything else the review team should know?

## Checklist

- [ ] Descriptive Pull Request title
- [ ] Link to relevant issues
- [ ] Provide necessary context for design reviewers
- [ ] Ensure test coverage is above agreed upon threshold
- [ ] Update documentation

katyasoup reviewed (approved):
LGTM! Only issue is the one with Firefox where it can `shift+tab` through the Query Repository table rows, but `tab` forward gets it trapped in the row (fine in Chrome/Safari)

robertandremitchell reviewed (commented):
some funky resizing when tabbing: 
<img width="832" alt="Screenshot 2025-07-02 at 3 48 00 PM" src="https://github.com/user-attachments/assets/b9a43d94-b570-4c03-aea5-9d83acde1712" />

fzhao99 commented:
> LGTM! Only issue is the one with Firefox where it can shift+tab through the Query Repository table rows, but tab forward gets it trapped in the row (fine in Chrome/Safari)

Think it had to do with the fact that the un-interactible row was being tabbable, which Firefox treated as an issue. Removed it and it seems ok.

> some funky resizing when tabbing:
Fixed!
